interview-questions.md

```diff
@@ -7,81 +7,86 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. Why would you use the method `super()`?
 
-  Your answer:
+  Your answer: The super() lets us call out to the parent function/method.
 
-  Researched answer:
+  Researched answer: When used in a constructor, the super keyword appears alone and must be used before the this keyword is used. The super keyword can also be used to call functions on a parent object.
 
 
 
 2. What is the DOM? What is the virtual DOM?
 
-  Your answer:
+  Your answer: Something to do with the computers connection to the internet?
 
-  Researched answer:
+  Researched answer: The HTML DOM (Document Object Model)
+When a web page is loaded, the browser creates a Document Object Model of the page.
+- The virtual DOM (VDOM) is a programming concept where an ideal, or “virtual”, representation of a UI is kept in memory and synced with the “real” DOM by a library such as ReactDOM. This process is called reconciliation.
 
 
 
 3. What is a higher-order function?
 
-  Your answer:
+  Your answer: A higher order function is a function that uses .map() and .filter() to condense the code, and make it easier to use, instead of for loops. 
 
-  Researched answer:
+  Researched answer: In a nutshell, a Higher-order function is a function that may receive a function as an argument and can even return a function. Higher-order functions are just like regular functions with an added ability of receiving and returning other functions are arguments and output.
 
 
 
 4. What is JSX?
 
-  Your answer:
+  Your answer: JavaScript run in html through react.
 
-  Researched answer:
+  Researched answer: It is called JSX, and it is a syntax extension to JavaScript. We recommend using it with React to describe what the UI should look like. JSX may remind you of a template language, but it comes with the full power of JavaScript.
+
+JSX produces React “elements”. We will explore rendering them to the DOM in the next section. Below, you can find the basics of JSX necessary to get you started.
 
 
 
 5. What is React?
 
-  Your answer:
+  Your answer: React is a pre built app that allows small chunks of code to be built and then put together piece by piece.
```

Comment:
Just about! I'd want to mention how its a library and that its used for creating .. *reactive* front-end applications. Your researched answer has it covered :)